core/src/main/scala/com/avast/cloud/metrics/datadog/impl/MetricFactoryImpl.scala

```diff
@@ -0,0 +1,26 @@
+package com.avast.cloud.metrics.datadog.impl
+
+import cats.effect.{ Clock, Sync }
+import com.avast.cloud.metrics.datadog.api.{ Gauge, Histogram, MetricFactory, UniqueSet }
+import com.timgroup.statsd.StatsDClient
+
+import scala.language.higherKinds
+
+class MetricFactoryImpl[F[_]: Sync](statsDClient: StatsDClient) extends MetricFactory[F] {
```

Comment:
so there is no way how to globally configure sample rate? That's IMHO the more common than configure it per aspect.